encoding/v2/kzg/prover/backend/icicle/multiframe_proof.go

```diff
@@ -290,7 +283,9 @@ func (c *KzgMultiProofBackend) twoEcnttOnDevice(
 	// we only allocate one large gpu memory for all operation, so it has to be large enough to cover all cases
 	// including the first and the second ECNTT
 	var bufferProjectivePointsOnDevice core.DeviceSlice
-	var secondECNTTDeviceSlice core.DeviceSlice
+	// free intermediate GPU memory
+	defer bufferProjectivePointsOnDevice.Free()
```

Comment:
Shouldn't we defer after malloc succeeds, otherwise we will call free on unitialized mem right